### PR TITLE
Fix tag parse language status item UI

### PR DESCRIPTION
**Pull request by michelleangela**
**GitHub issue:**
https://github.com/microsoft/vscode-cpptools/issues/10749

**Note:** this has a corresponding language server changeset.

**Changes:**

- Fixed updating states/status of tag parse due to missing conditions or unhandled cases.
- Remove code that was used for old UI.
- Grouped functions and objects into their respective feature status (i.e. code analysis status vs IntelliSense status).
- Refactored code to remove redundancies.

**Cases tested:**

- startup with no database
- startup with database complete
- startup with database incomplete
- reset database
- pause/resume during workspace parsing
- pause/resume during workspace and open file parsing
- parsing open files only
- parsing workspace only
- parsing open files and workspace
- rescan workspace after parse complete

### CODE REVIEW

File: Extension/src/LanguageServer/ui.ts [hunk 1]
```diff
@@ -58,288 +58,266 @@ const commandArguments: string[] = []; // We report the sender of the command
...
-        this.browseEngineStatusItem.detail = localize("cpptools.detail.tagparser", "Initializing...");
-        this.browseEngineStatusItem.text = "$(database)";
-        this.browseEngineStatusItem.command = {
-            command: "C_Cpp.RescanWorkspace",
-            title: this.workspaceRescanText,
+    private flameTimeout?: NodeJS.Timeout;
+    private setIsUpdatingIntelliSense(val: boolean): void {
+        const settings: CppSettings = new CppSettings((vscode.workspace.workspaceFolders && vscode.workspace.workspaceFolders.length > 0) ? vscode.workspace.workspaceFolders[0]?.uri : undefined);
+
+        if (settings.intelliSenseEngine === "disabled") {
```
Comment (bobbrow) [inline Extension/src/LanguageServer/ui.ts:138]: Did we determine whether this was dead code?  I was thinking that we don't enable the language status when IntelliSense is disabled.

File: Extension/src/LanguageServer/ui.ts [hunk 2]
```diff
@@ -58,288 +58,266 @@ const commandArguments: string[] = []; // We report the sender of the command
...
+        this.ShowConfiguration = true;
+
+        this.configureIntelliSenseStatusBarItem = this.createConfigureIntelliSenseStatusBarItem();
         void this.ShowConfigureIntelliSenseButton(false, this.currentClient);
+    }
 
-        this.intelliSenseStatusItem = vscode.languages.createLanguageStatusItem(`cpptools.status.${LanguageStatusPriority.Mid}.intellisense`, util.documentSelector);
-        this.intelliSenseStatusItem.name = localize("cpptools.status.intellisense", "C/C++ IntelliSense Status");
-        this.intelliSenseStatusItem.text = this.idleIntelliSenseText;
+    //#region IntelliSense language status
```
Comment (sean-mcmanus) [inline Extension/src/LanguageServer/ui.ts:126]: Is this supposed to be commented out? Or is that the syntax for regions?

**Review by sean-mcmanus — DISMISSED** [review]
I haven't tested this, but I plan to later.

File: Extension/src/LanguageServer/ui.ts (hunk 2 above)
Comment (michelleangela) [reply Extension/src/LanguageServer/ui.ts:126]: That's the syntax for regions in Typescript. By typing "#region" it automatically puts "//".

**Review by sean-mcmanus — DISMISSED** [review]

**Review by sean-mcmanus — APPROVED** [review]

**Pull request merged**